### PR TITLE
[DUOS-2059][risk=no] SendGrid Refactor

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Partially addresses https://broadworkbench.atlassian.net/browse/DUOS-2059

This PR makes some minor, but extensive, changes to how we send messages via SendGrid. This is already a large PR, so I am hoping to move this through before tackling the original goal of capturing SendGrid response info per email we send.

### Changes:
* Renamed the `MailService` class to `SendGridAPI` for clarity. Unfortunately, git didn't retain the rename and considers it a +/-
* Migrated all uses of sending multiple messages to sending a single message
  * I want to do this so that we can more easily track what was sent via SendGrid per message
* Lots of message and test refactoring to accomplish ☝🏽 

---
A future PR will:
* Make use of the returned SendGrid `Response` 
* Improve testing of `SendGridAPI` (the actual `SendGrid` object is not mockable in the current implementation)
* Save a SendGrid response per email entity

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/mail/SendGridAPI.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    public SendGridAPI(MailConfiguration config) {
+        this.fromAccount = config.getGoogleAccount();
+        this.sendGrid = new SendGrid(config.getSendGridApiKey());
+        this.activateEmailNotifications = config.isActivateEmailNotifications();
+    }
+
+    private Optional<Response> sendMessage(Mail message) {
+        if (activateEmailNotifications) {
+            try {
+                // See https://github.com/sendgrid/sendgrid-java/issues/163
```
Comment (JVThomas) [inline src/main/java/org/broadinstitute/consent/http/mail/SendGridAPI.java:58]: Getting a `Page Not Found` when trying to access this link

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/mail/SendGridAPI.java:58]: Yeah, I think they disabled public view of their issues page. This is copied from the previous class and unrelated to the refactor, so I left it in place until I can find a solution.

**Review by JVThomas — APPROVED** [review]
Code looks good to me

**Review by connorlbark — APPROVED** [review]
👍 I like this refactor a lot, as someone who recently dove in to learn about this side of the codebase, this is much less confusing and more straightforward now.

**Pull request merged**